woocommerce/grow
# Add eslint and JavaScript linting in GitHub Actions

### Changes proposed in this Pull Request:

Implements parts of #4 

- Add `eslint`, related deps and their configurations.
- Add a GitHub Actions workflow to run JavaScript linting.
- Fix all linting errors and warnings.

Before merging this PR:

- [x] Wait for #35 to be released.
- [x] Change `actions-v1-pre` to `actions-v1`.

### Detailed test instructions:

1. `npm install`
2. `npm run lint:js` locally to test the eslint
   - 💡 It may gets some `import/no-unresolved` warnings because the *node_modules* directories in sub-packages don't exist. Those warnings will disappear after running `npm install` in those sub-packages.
3. View the [check](https://github.com/woocommerce/grow/runs/7320238300?check_suite_focus=true) that was intentionally left with linting errors and warnings.
4. View the annotations attached to this PR:
   ![github com_woocommerce_grow_pull_36_files](https://user-images.githubusercontent.com/17420811/178727568-387c2934-aab7-4222-a5dd-1dd467fb6177.png)
5. View the Check tab of this PR to see the successful linting check.
